### PR TITLE
Update docker.yml

### LINKED ISSUE #726: Docker Github Action
## Summary of Bug

https://github.com/cosmos/gaia/blob/main/.github/workflows/docker.yml

Also saw a "wrong password" error in a build job.  

Docker image uses Alpine, which makes great multiplatform builds, so we should specify that in the action and get those, as well. 

Looks like it is actually pushing to:

https://hub.docker.com/r/tendermint/gaia/tags?page=1&ordering=last_updated

## Version

dc11e61d815bf294a79c558e7b4a04d804d8daff

## Steps to Reproduce

Try to find where the Docker Github action pushes to.  

## Proposed Solution

Approximately:

```yaml
name: Publish Docker image
on:
  # Trigger the workflow on push or pull request,
  # but only for the main branch
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
  release:
    types: # This configuration does not affect the page_build event above
      - created
jobs:
  push_to_registry:
    name: Push Multiplatform Gaia Docker image to GitHub Packages
    runs-on: ubuntu-latest
    steps:
      - name: Check out the repo
        uses: actions/checkout@v2

      - name: Set up QEMU
         run: docker run --rm --privileged multiarch/qemu-user-static --reset -p yes --credential yes

      - name: Login to Packages Registry
        uses: docker/login-action@v1 
        with:
          username: ${{ github.actor }}
          password: ${{ secrets.GITHUB_TOKEN }}

      - name: Push to GitHub Packages
        uses: docker/build-push-action@v2
        with:
          context: .
          platforms: linux/amd64,linux/arm64
          registry: docker.pkg.github.com
          repository: ${{ github.actor }}/gaia/gaia
          tags: ${{ GITHUB_REPOSITORY }}:${GITHUB_REF##*/},${{ GITHUB_REPOSITORY }}:${GITHUB_REF},
```

This provides:

* Something that can be forked into other repositories and still build
* Docker images by branch and commit hash

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by faddat**
Closes: #726 

## Description

Multiplatform docker image for Gaia.

Note:  Previously this said it was building simapp, but it did not look like it is doing that anymore. 
______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/gaia/blob/main/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/main/README.md#merging-a-pr))

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -1,55 +1,39 @@
...
+      - name: Push to GitHub Packages
         uses: docker/build-push-action@v2
         with:
-          push: ${{ github.event_name != 'pull_request' }}
-          tags: ${{ steps.prep.outputs.tags }}
+          context: .
+          platforms: linux/amd64,linux/arm64
+          registry: docker.pkg.github.com
+          repository: ${{ github.actor }}/gaia/gaia
+          tags: ${GITHUB_REPOSITORY}:${BRANCH_NAME},${GITHUB_REPOSITORY}:${BRANCH_NAME}
```
Comment (tac0turtle) [inline .github/workflows/docker.yml]: does this work for releases?

Comment (faddat) [reply .github/workflows/docker.yml]: Not yet, actually it should have contained a comment/todo:

* Need to tag releases.

Comment (faddat) [reply .github/workflows/docker.yml]: 'bout an hour later, I guess that it'll work for releases

**Comment by faddat** [comment]
Propose that we leave this unmerged for a week or two until GitHub lets us login to ghcr.io with secrets.GITHUB_TOKEN

**Comment by faddat** [comment]
I'm going to close this, and just change the confusing comments in the code :).

**Pull request closed without merging**